File: client-scalaz/src/main/scala/org/openapitools/client/api/BlockedNotificationResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import BlockedNotificationResponse._

case class BlockedNotificationResponse (
  blockedNotificationId: Option[Long],
created: Option[Long],
event: Option[String],
conduit: Option[String],
customType: Option[String],
`contentType`: Option[String],
contentId: Option[Long],
searchTags: Option[String],
blocked: Option[Boolean],
deleted: Option[Boolean])

object BlockedNotificationResponse {
  import DateTimeCodecs._

  implicit val BlockedNotificationResponseCodecJson: CodecJson[BlockedNotificationResponse] = CodecJson.derive[BlockedNotificationResponse]
  implicit val BlockedNotificationResponseDecoder: EntityDecoder[BlockedNotificationResponse] = jsonOf[BlockedNotificationResponse]
  implicit val BlockedNotificationResponseEncoder: EntityEncoder[BlockedNotificationResponse] = jsonEncoderOf[BlockedNotificationResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/blocked_notification_response.clj
(ns sirqul-io-t-platform.specs.blocked-notification-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def blocked-notification-response-data
  {
   (ds/opt :blockedNotificationId) int?
   (ds/opt :created) int?
   (ds/opt :event) string?
   (ds/opt :conduit) string?
   (ds/opt :customType) string?
   (ds/opt :contentType) string?
   (ds/opt :contentId) int?
   (ds/opt :searchTags) string?
   (ds/opt :blocked) boolean?
   (ds/opt :deleted) boolean?
   })

(def blocked-notification-response-spec
  (ds/spec
    {:name ::blocked-notification-response
     :spec blocked-notification-response-data}))
